meson.build

```diff
@@ -91,12 +91,20 @@ nanoarrow_lib = library(
     gnu_symbol_visibility: 'hidden',
 )
 
+pkg = import('pkgconfig')
+pkg.generate(
+    nanoarrow_lib,
+    description: 'Helpers for Arrow C Data & Arrow C Stream interfaces',
+)
+
 nanoarrow_dep = declare_dependency(
     include_directories: [incdir],
     link_with: nanoarrow_lib,
     compile_args: nanoarrow_dep_args,
 )
 
+meson.override_dependency('nanoarrow', nanoarrow_dep)
```

Comment:
In today's wrap we have a provides section that looks like:

```ini
[provides]
nanoarrow = nanoarrow_dep
nanoarrow_ipc = nanoarrow_ipc_dep
```

But since Meson version 0.54.0, they have advised that a project internally call `meson.override_dependency` instead, so that the project has total control over their variable naming. As such, on the next release of Meson, we can simplify our wrap entry to just:

```ini
[provides]
dependency_names = [nanoarrow, nanoarrow-ipc, ...]
```

and decouple the internal variable name from the wrap system